app/src/main/java/com/andrew/tasky/presentation/event_detail/EventDetailFragment.kt

```diff
@@ -0,0 +1,723 @@
+
+        supportFragmentManager.setFragmentResultListener(
+            "REQUEST_KEY",
+            viewLifecycleOwner
+        ){
+                resultKey, bundle -> if(resultKey == "REQUEST_KEY"){
+            val date = bundle.getString("SELECTED_DATE")
+            val formatter = DateTimeFormatter.ofPattern("MMM dd yyyy")
+
+            if(view == startTimeDateBinding.dateTextView || view == startTimeDateBinding.dateButton){
+                viewModel.setStartDate(LocalDate.parse(date, formatter))
+            }
+            else if(view == endTimeDateBinding.dateTextView || view == endTimeDateBinding.dateButton){
+                viewModel.setEndDate(LocalDate.parse(date, formatter))
+            }
+        }
+        }
+        datePickerFragment.show(supportFragmentManager, "DatePickerFragment")
+    }
+
+    //View parameter determines which view popup appears under
+    private fun showReminderOptionsPopupMenu(view: View){
+        val popupMenu = PopupMenu(requireContext(), view)
+        popupMenu.inflate(R.menu.menu_reminder_time_options)
+        reminderLayoutBinding.reminderTextView.apply {
+            popupMenu.setOnMenuItemClickListener {
+                when (it.itemId) {
+                    R.id.tenMinutes -> {
+                        viewModel.setSelectedReminderTime(ReminderTimes.TEN_MINUTES_BEFORE)
+                        true
+                    }
+                    R.id.thirtyMinutes -> {
+                        viewModel.setSelectedReminderTime(ReminderTimes.THIRTY_MINUTES_BEFORE)
+                        true
+                    }
+                    R.id.oneHour -> {
+                        viewModel.setSelectedReminderTime(ReminderTimes.ONE_HOUR_BEFORE)
+                        true
+                    }
+                    R.id.sixHours -> {
+                        viewModel.setSelectedReminderTime(ReminderTimes.SIX_HOURS_BEFORE)
+                        true
+                    }
+                    R.id.oneDay -> {
+                        viewModel.setSelectedReminderTime(ReminderTimes.ONE_DAY_BEFORE)
+                        true
+                    }
+                    else -> true
+                }
+            }
+        }
+        popupMenu.show()
+    }
+
+    private fun showDeleteConfirmationDialogFragment(){
+        val deleteConfirmationDialogFragment = DeleteConfirmationDialogFragment()
+        val supportFragmentManager = requireActivity().supportFragmentManager
+
+        supportFragmentManager.setFragmentResultListener(
+            "REQUEST_KEY",
+            viewLifecycleOwner
+        ){
+                resultKey, bundle -> if(resultKey == "REQUEST_KEY"){
+            val deleteAgendaItem = bundle.getBoolean("DELETE_AGENDA_ITEM")
+            if (deleteAgendaItem){
+                args.agendaItem?.let {
+                    AgendaItems.deleteAgendaItem(it)
+                }
+                navController.popBackStack()
+            }
+        }
+        }
```

Comment:
Here, the formatting looks quite messed up, I'd recommend taking a look at setting up a linting & auto-formatting tool like ktlint to enforce a consistent style. Here I explain it:
https://www.youtube.com/watch?v=hSgPNbEcX98